### PR TITLE
Support user contexts for RBAC

### PR DESCRIPTION
**Pull request by joshwget**
rancher/rancher#8717

### CODE REVIEW

File: package/kubectl-shell.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+set -ex
```
Comment (ibuildthecloud) [inline package/kubectl-shell.sh]: make sure you remove -x before merging.  same with other script.

File: package/kubectl-shell.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+set -e
+
+CONTEXT_DIR=/var/lib/kubectl-contexts/$(uuidgen -r)
+mkdir -p ${CONTEXT_DIR}
+mount --bind ${CONTEXT_DIR} ${CONTEXT_DIR}
```
Comment (ibuildthecloud) [inline package/kubectl-shell.sh]: What's the point of this mount?  Will it ever get unmounted?

Comment (joshwget) [inline package/kubectl-shell.sh]: Seemed necessary based on the docs for `unshare`, will remove and see if it's actually needed.

**Comment by ibuildthecloud** [comment]
LGTM

**Pull request merged**